cmd/juju/cloud/add_test.go

```diff
@@ -148,83 +182,146 @@ func (*addSuite) TestAddBadCloudName(c *gc.C) {
 
 func (*addSuite) TestAddExisting(c *gc.C) {
 	fake := newFakeCloudMetadataStore()
-	fake.Call("ParseCloudMetadataFile", "fake.yaml").Returns(homestackMetadata(), nil)
+	mockStore := &mockCloudeMetadataStore{}
+
+	cloudFile := prepareTestCloudYaml(c, homeStackYamlFile)
+	defer cloudFile.Close()
+	defer os.Remove(cloudFile.Name())
+
+	mockCloud, err := mockStore.ParseCloudMetadataFile(cloudFile.Name())
```

Comment:
We don't need mockCloudMetadataStore
We can just call cloudfile.ParseCloudMetadataFile(cloudFile.Name()) directly.